### PR TITLE
refactor: migrate prefer last

### LINKED ISSUE #279: Migrate prefer_last

### PR DESCRIPTION
**Pull request by andrew-bekhiet-solid**
Closes #279

### CODE REVIEW

File: lib/src/lints/prefer_last/fixes/prefer_last_fix.dart [hunk 1]
```diff
@@ -1,67 +1,70 @@
...
-      return expression.isCascaded
-          ? '..last'
-          : '${expression.target ?? ''}.last';
-    } else if (expression is IndexExpression) {
-      return expression.isCascaded
-          ? '..last'
-          : '${expression.target ?? ''}.last';
-    } else {
-      return '.last';
+    switch (expression) {
```
Comment (danylo-safonov-solid) [inline lib/src/lints/prefer_last/fixes/prefer_last_fix.dart:46]: should we also add test cases for cascades and null-awares, I think that only makes sense?

**Review by danylo-safonov-solid — CHANGES_REQUESTED** [review]

File: test/prefer_last/prefer_last_rule_test.dart
Comment (solid-danylosafonov) [inline test/prefer_last/prefer_last_rule_test.dart:1]: let's update this file to use `expectLint`

File: lib/src/lints/prefer_last/visitors/prefer_last_visitor.dart [hunk 2]
```diff
@@ -34,41 +37,76 @@ class PreferLastVisitor extends RecursiveAstVisitor<void> {
...
+      ):
+        return _unwrapExpression(operand);
 
-    return null;
+      case ParenthesizedExpression(:final expression):
+        return _unwrapExpression(expression);
+
+      default:
+        return expression;
+    }
```
Comment (solid-danylosafonov) [inline lib/src/lints/prefer_last/visitors/prefer_last_visitor.dart]:
```dart
   return switch (expression) {
      PostfixExpression(
        :final operand,
        operator: Token(type: TokenType.BANG),
      ) => _unwrapExpression(operand),
      ParenthesizedExpression(:final expression) => _unwrapExpression(expression),
      _ => expression,
    };
```

**Review by solid-danylosafonov — APPROVED** [review]
